### ISSUE TITLE
Masking in Embedding layer

### ISSUE THREAD

**Issue by AakashKumarNain** [patrick-kidger/equinox#668]
When the input sequences are of different lengths, padding with zeros is a common thing to make the shape of every sample in a batch of the same size. But these zeros shouldn't be accounted for operations down the line or while gradient updates as they don't represent an actual token in the sequence. 

Given that embedding layer doesn't do it explicitly (unless I missed something in the documentation), is there any example I can look up to as the recommended workflow?

**Comment by patrick-kidger** [comment]
The frequent thing to do here is to do padding before the embedding, and include your chosen 'pad token' (whether that's a zero or anything else) in the embedding layer itself.

For example here's a very simple single-character tokeniser:
```python
alphabet = {'a': 0, 'b': 1, 'c': 2, ... 'z': 25, ' ': 26, '<pad>': 27}

# num_embeddings ncludes padding
embedding = eqx.nn.Embedding(num_embeddings=len(alphabet), embedding_size=...)

input = "the quick brown fox"
length = 1024
x = np.full((length,), alphabet['<pad>'])
for i, char in enumerate(input):
    x[i] = alphabet[char]

x = embedding(x)
```
